Docs Patch to include Kubernetes Installation

david-yu opened the pull request:
This patch includes a refactor of the current Install docs to highlight that Consul can be installed on Kubernetes. Open to feedback as it refactors some of the layout of the previous install page.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/consul?pullRequest=7075)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/consul?pullRequest=7075)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/consul?pullRequest=7075).</sub>

kaitlincarter-hc commented on website/source/docs/install/index.html.md:
```diff
@@ -9,19 +9,55 @@ description: |-
+1. [Installing on Kubernetes](#install-on-kubernetes) 
 
-1. Installing [from source](#compiling-from-source)
+1. Installing as a standalone binary 
```
Could you add the link back? `(#precompiled-binaries)`

david-yu replied:
This is included as a sub bullet link below.

kaitlincarter-hc commented:
<img width="719" alt="Screen Shot 2020-01-17 at 10 19 57 AM" src="https://user-images.githubusercontent.com/43049322/72627967-02103580-3913-11ea-8f81-2e8d28126eb5.png">

It looks like the formatting is off. We can probably remove point a and b too.

kaitlincarter-hc commented on website/source/docs/install/index.html.md:
```diff
@@ -9,19 +9,55 @@ description: |-
 # Install Consul
 
-Installing Consul is simple. There are two approaches to installing Consul:
+Installing Consul is simple. There are two main approaches to installing Consul:
```
```suggestion
There are two main approaches for installing Consul:
```

kaitlincarter-hc commented on website/source/docs/install/index.html.md:
```diff
@@ -9,19 +9,55 @@ description: |-
+   b. Installing [from source](#compiling-from-source)
 
-## Precompiled Binaries
+   Downloading a precompiled binary is easiest, and we provide downloads over TLS
```
```suggestion
   Downloading a precompiled binary is quickest and we provide downloads over TLS
```

kaitlincarter-hc commented on website/source/docs/install/index.html.md:
```diff
@@ -9,19 +9,55 @@ description: |-
+
+## Install on Kubernetes
+
+Installing Consul on Kubernetes is done through Helm, which provides an easy way to deploy applications on Kubernetes. 
```
```suggestion
Installing Consul on Kubernetes is done through Helm, which provides a convenient way to deploy applications on Kubernetes. 
```

kaitlincarter-hc commented on website/source/docs/install/index.html.md:
```diff
@@ -9,19 +9,55 @@ description: |-
+Installing Consul on Kubernetes is done through Helm, which provides an easy way to deploy applications on Kubernetes. 
+
+Determine the latest version of the Consul Helm chart
+by visiting [https://github.com/hashicorp/consul-helm/releases](https://github.com/hashicorp/consul-helm/releases).
```
We don't typically use full urls in docs.

david-yu replied:
This was copied and pasted from the Kubernetes install docs, how would you link to a Github releases page from this doc?

lkysow replied:
We shouldn't be copying this doc. Just link to the existing install guide. Otherwise we have to update things in multiple places which inevitably we'll forget to do.

kaitlincarter-hc commented on website/source/docs/install/index.html.md:
````diff
@@ -9,19 +9,55 @@ description: |-
+...
+```
+
+A more detailed set of instructions can be found [here](/docs/platform/k8s/run.html). 
````
Can you link to both the docs and Learn? 
```suggestion
A more detailed set of instructions can be found [here](/docs/platform/k8s/run.html). 

```

david-yu commented:
@kaitlincarter-hc Just addressed all the comments previously, this is a little more involved than @lkysow's PR that included a link to the Kubernetes install docs. 

How would you like to proceed here?

lkysow commented:
You need to rebase this based on the latest changes. We added a link to install kube this weekend.

david-yu commented:
Going to close this as we discussed reordering the preference for install would require input from the entire consul team. Also install instructions may change in the future to include namespaces, so a link would be better for now.
